sellercenter-product-port/src/main/java/com/gfg/sellercenter/product/port/service/ProductReaderService.java

```diff
@@ -53,23 +55,18 @@ public Product getById(int productId) throws IOException {
             throw new IllegalArgumentException("Product ids can not be empty");
         }
 
-        String requestUrl = hostUrl + API_PATH + "?product_ids=[" + joinIds(productIds) + "]";
-
-        JSONArray products = jsonReader.readAll(requestUrl);
-
         Map<Integer, Product> result = new HashMap<>();
+        String requestUrl = hostUrl + API_PATH + "?product_ids=[" + joinIds(productIds) + "]";
+        JsonParser parser = productMapper.getFactory().createParser(new URL(requestUrl));
```

Comment:
To still have unit test for it, I will need to mock `productMapper`. Do you maybe have any ideas how to do it?